cli/pkg/utils/utils.go

```diff
@@ -0,0 +1,310 @@
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+const REPO_LOCATION = "https://charts.timescale.com"
+const DEFAULT_CHART = "timescale/tobs"
+
+func AddTobsHelmChart() error {
+	w := io.Writer(os.Stdout)
+	addchart := exec.Command("helm", "repo", "add", "timescale", REPO_LOCATION)
+	addchart.Stdout = w
+	addchart.Stderr = w
+	fmt.Println("Adding Timescale Helm Repository")
+	err := addchart.Run()
+	if err != nil {
+		return fmt.Errorf("could not install The Observability Stack: %w", err)
+	}
+	return err
+}
+
+func UpdateTobsHelmChart(upgrade bool) error {
+	update := exec.Command("helm", "repo", "update")
+
+	// if upgrade flow hide the
+	// stdout of helm repo update
+	if !upgrade {
+		w := io.Writer(os.Stdout)
+		update.Stdout = w
+		update.Stderr = w
+	}
+
+	fmt.Println("Fetching updates from repository")
+	err := update.Run()
+	if err != nil {
+		return fmt.Errorf("could not install The Observability Stack: %w", err)
+	}
+	return err
+}
+
+type ChartMetadata struct {
+	APIVersion   string `yaml:"apiVersion"`
+	AppVersion   string `yaml:"appVersion"`
+	Dependencies []struct {
+		Condition  string `yaml:"condition"`
+		Name       string `yaml:"name"`
+		Repository string `yaml:"repository"`
+		Version    string `yaml:"version"`
+	} `yaml:"dependencies"`
+	Description string `yaml:"description"`
+	Name        string `yaml:"name"`
+	Type        string `yaml:"type"`
+	Version     string `yaml:"version"`
+}
+
+type DeployedChartMetadata struct {
+	Name       string `json:"name"`
+	Namespace  string `json:"namespace"`
+	Revision   string `json:"revision"`
+	Updated    string `json:"updated"`
+	Status     string `json:"status"`
+	Chart      string `json:"chart"`
+	AppVersion string `json:"app_version"`
+}
+
+// struct for comparing values.yaml files
+type TobsValues struct {
```

Comment:
This won't work because it only compares default properties in the yaml, but the subcharts have lots of fields that can be added